pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -145,6 +158,12 @@ def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule"):
         optimizers = trainer.optimizers
         lr_schedulers = trainer.lr_schedulers
 
+        sharded_plugins = [DDPFullyShardedPlugin, DeepSpeedPlugin]
+        if FullyShardedDataParallel:
+            sharded_plugins.append(FullyShardedDataParallel)
+        if isinstance(trainer.training_type_plugin, tuple(sharded_plugins)):
+            raise MisconfigurationException("SWA does not currently support sharded models.")
```

Comment:
FullyShardedDataParallel is not a plugin :) 

```suggestion
        if isinstance(trainer.training_type_plugin, (DDPFullyShardedPlugin, DeepSpeedPlugin):
            raise MisconfigurationException("SWA does not currently support sharded models.")
```

You will also have to include these:
DDPShardedPlugin, DDPSpawnShardedPlugin

Reply:
I think DDPShardedPlugin and DDPSpawnShardedPlugin are actually OK. At least from my quick testing they seem to work with SWA. It's only when the model parameters are sharded that we get problems, but sharding the optimizer and gradients seems fine. DeepSpeed is a bit different as it looks like parameter sharding is optional so maybe it could be made to work fairly easily, but with the default deepspeed strategy the SWA callback raises an exception due to there not being 1 optimizer in on_fit_start.